cpp/include/cudf/hashing/detail/murmurhash3_x86_32.cuh

```diff
@@ -16,165 +16,91 @@
-// https://github.com/aappleby/smhasher/blob/master/src/MurmurHash3.cpp
-//-----------------------------------------------------------------------------
-// MurmurHash3 was written by Austin Appleby, and is placed in the public
-// domain. The author hereby disclaims copyright to this source code.
-// Note - The x86 and x64 versions do _not_ produce the same results, as the
-// algorithms are optimized for their respective platforms. You can still
-// compile and run any of them on any platform, but your performance with the
-// non-native version will be less than optimal.
 template <typename Key>
-struct MurmurHash3_x86_32 {
-  using result_type = hash_value_type;
-
-  constexpr MurmurHash3_x86_32() = default;
-  constexpr MurmurHash3_x86_32(uint32_t seed) : m_seed(seed) {}
-
-  [[nodiscard]] __device__ inline uint32_t fmix32(uint32_t h) const
-  {
-    h ^= h >> 16;
-    h *= 0x85ebca6b;
-    h ^= h >> 13;
-    h *= 0xc2b2ae35;
-    h ^= h >> 16;
-    return h;
-  }
-
-  [[nodiscard]] __device__ inline uint32_t getblock32(std::byte const* data,
-                                                      cudf::size_type offset) const
-  {
-    // Read a 4-byte value from the data pointer as individual bytes for safe
-    // unaligned access (very likely for string types).
-    auto const block = reinterpret_cast<uint8_t const*>(data + offset);
-    return block[0] | (block[1] << 8) | (block[2] << 16) | (block[3] << 24);
-  }
-
-  [[nodiscard]] result_type __device__ inline operator()(Key const& key) const
-  {
-    return compute(normalize_nans_and_zeros(key));
-  }
+struct MurmurHash3_x86_32 : public cuco::murmurhash3_32<Key> {
+  using result_type = typename cuco::murmurhash3_32<Key>::result_type;
 
-  template <typename T>
-  result_type __device__ inline compute(T const& key) const
+  __host__ __device__ constexpr MurmurHash3_x86_32(uint32_t seed = cudf::DEFAULT_HASH_SEED)
+    : cuco::murmurhash3_32<Key>{seed}
   {
-    return compute_bytes(reinterpret_cast<std::byte const*>(&key), sizeof(T));
   }
 
-  result_type __device__ inline compute_remaining_bytes(std::byte const* data,
-                                                        cudf::size_type len,
-                                                        cudf::size_type tail_offset,
-                                                        result_type h) const
+  __device__ result_type operator()(Key const& key) const
   {
-    // Process remaining bytes that do not fill a four-byte chunk.
-    uint32_t k1 = 0;
-    switch (len % 4) {
-      case 3: k1 ^= std::to_integer<uint8_t>(data[tail_offset + 2]) << 16; [[fallthrough]];
-      case 2: k1 ^= std::to_integer<uint8_t>(data[tail_offset + 1]) << 8; [[fallthrough]];
-      case 1:
-        k1 ^= std::to_integer<uint8_t>(data[tail_offset]);
-        k1 *= c1;
-        k1 = rotate_bits_left(k1, rot_c1);
-        k1 *= c2;
-        h ^= k1;
-    };
-    return h;
+    return cuco::murmurhash3_32<Key>::operator()(key);
```

Comment:
Is there a better way to write this so that we "just call the call operator" instead of using `::operator()`? Does this work?
```suggestion
    return cuco::murmurhash3_32<Key>{}(key);
```

Reply:
Ah. Is this a case where composition would be better than inheritance? https://en.wikipedia.org/wiki/Composition_over_inheritance

Reply:
> Is this a case where composition would be better than inheritance?

Great question! I'd love to hear @lamarrr's insights on this.

Reply:
The current approach looks okay to me, although I'd prefer having it as a member of the hasher if we were going for composition. It's purely stylistic in this case and gives no semantic difference.